src/java.base/share/classes/java/lang/runtime/StringTemplateImplFactory.java

```diff
@@ -0,0 +1,197 @@
+        }
+        interpolateMH = MethodHandles.filterArguments(interpolateMH, 0, components);
+        mt = MethodType.methodType(String.class, StringTemplateImpl.class);
+        interpolateMH = MethodHandles.permuteArguments(interpolateMH, mt, permute);
+
+        mt = MethodType.methodType(List.class, ptypes);
+        MethodHandle valuesMH = TO_LIST.asCollector(Object[].class, components.length).asType(mt);
+        valuesMH = MethodHandles.filterArguments(valuesMH, 0, components);
+        mt = MethodType.methodType(List.class, StringTemplateImpl.class);
+        valuesMH = MethodHandles.permuteArguments(valuesMH, mt, permute);
+
+        MethodHandle constructor = MethodHandles.insertArguments(CONSTRUCTOR,0,
+                elements.primitiveCount(), elements.objectCount(),
+                fragments, valuesMH, interpolateMH);
+        constructor = MethodHandles.foldArguments(elements.initializer(), 0, constructor);
+
+        mt = MethodType.methodType(StringTemplate.class, ptypes);
+        constructor = constructor.asType(mt);
+
+        return constructor;
+    }
+
+    /**
+     * Generic {@link StringTemplate}.
+     *
+     * @param fragments  immutable list of string fragments from string template
+     * @param values     immutable list of expression values
+     */
+    private record SimpleStringTemplate(List<String> fragments, List<Object> values)
+            implements StringTemplate {
+        @Override
+        public String toString() {
+            return StringTemplate.toString(this);
+        }
+    }
+
+    /**
+     * Returns a new StringTemplate composed from fragments and values.
+     *
+     * @param fragments array of string fragments
+     * @param values    array of expression values
+     *
+     * @return StringTemplate composed from fragments and values
+     */
+    static StringTemplate newStringTemplate(String[] fragments, Object[] values) {
+        return new SimpleStringTemplate(List.of(fragments), toList(values));
+    }
+
+    /**
+     * Returns a new StringTemplate composed from fragments and values.
+     *
+     * @param fragments list of string fragments
+     * @param values    array of expression values
+     *
+     * @return StringTemplate composed from fragments and values
+     */
+    static StringTemplate newStringTemplate(List<String> fragments, Object[] values) {
+        return new SimpleStringTemplate(List.copyOf(fragments), toList(values));
+    }
+
+    /**
+     * Returns a new StringTemplate composed from fragments and values.
+     *
+     * @param fragments list of string fragments
+     * @param values    list of expression values
+     *
+     * @return StringTemplate composed from fragments and values
+     */
+    static StringTemplate newStringTemplate(List<String> fragments, List<?> values) {
+        return new SimpleStringTemplate(List.copyOf(fragments), toList(values.stream().toArray()));
```

Comment:
```suggestion
        return new SimpleStringTemplate(List.copyOf(fragments), values.stream().toList());
```
per https://github.com/openjdk/jdk/pull/10889#discussion_r1146455491

Reply:
Changed